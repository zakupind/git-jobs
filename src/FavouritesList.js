import React from 'react';
import {connect} from 'react-redux';

import FavouritesItem from './Favourites Item'

import { checkFavorites } from  './REDUX/action';

class FavouritesList extends React.Component {
    componentDidMount() {
        this.props.checkFavorites()
    }

    render() {
        return(
            <div>
                <h2>Избранное</h2>
                <ul>
                    {this.props.favouritesList.map(item => (
                        <FavouritesItem key={item.id} {...item}/>
                    ))}
                </ul>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    favouritesList: state.favourites.favouritesList
})

const mapDispatchToProps = {
    checkFavorites
}

export default connect(mapStateToProps, mapDispatchToProps)(FavouritesList);