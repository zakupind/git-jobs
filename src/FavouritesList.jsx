import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import FavouritesItem from './FavouritesItem';

function FavouritesList(props) {
  const { favouritesList } = props;
  return (
    <div className="favourites-list">
      <h2 className="favourites-list__title">Избранное</h2>
      <ul className="favourites-list__list">
        {favouritesList.length === 0
          ? <span>Нет избранных вакансий</span>
          : Object.keys(favouritesList).map((key) => (
            <FavouritesItem key={key} {...favouritesList[key]} />))}
      </ul>
    </div>
  );
}

const mapStateToProps = (state) => ({
  favouritesList: state.favourites.favouritesList,
});

FavouritesList.propTypes = {
  favouritesList: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

export default connect(mapStateToProps)(FavouritesList);
