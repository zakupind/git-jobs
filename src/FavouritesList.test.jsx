import enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { FavouritesList } from './FavouritesList';

enzyme.configure({ adapter: new Adapter() });

describe('Тест компонента листа избранных вакансий', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      favouritesList: {
        id1:
      {
        id: 'id1',
        title: 'title1',
        url: 'url1',
      },
        id2:
      {
        id: 'id2',
        title: 'title2',
        url: 'url2',
      },
      },
    };

    component = shallow(<FavouritesList {...props} />);
  });

  test('Проверка на render компонента', () => {
    expect(component).toBeTruthy();
  });

  test('render FavouriteList', () => {
    expect(component).toMatchSnapshot();
  });
});
