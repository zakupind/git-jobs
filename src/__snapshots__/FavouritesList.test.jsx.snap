// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента листа избранных вакансий render FavouriteList 1`] = `
<div
  className="favourites-list"
>
  <h2
    className="favourites-list__title"
  >
    Избранное
  </h2>
  <ul
    className="favourites-list__list"
  >
    <Connect(FavouritesItem)
      id="id1"
      key="id1"
      title="title1"
      url="url1"
    />
    <Connect(FavouritesItem)
      id="id2"
      key="id2"
      title="title2"
      url="url2"
    />
  </ul>
</div>
`;
